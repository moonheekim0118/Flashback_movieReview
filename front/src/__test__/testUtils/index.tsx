import 'jsdom-global/register'; 
import React from 'react';
import { mount } from 'enzyme';

const TestHook = ({ callback }) => {
  callback();
  return null;
};

export const testHook = (callback) => {
  return mount(<TestHook callback={callback} />);
};


export const MyInfoProp = 
{ id:'test',
nickname:'test',
Reviews:1,
proilePic:""
}  

export const MovieProp =  
{ id:'test', 
title:'test', 
image:'test', 
director:'test', 
pubDate:'test'};

export const AuthorProp = { id:'test', nickname:'test'};

export const ReviewProp = 
{ id:'test', 
Movie: MovieProp, 
User:AuthorProp, 
rating:'GOOD', 
shortComment:'test',
character:'test',
line:'test',
scene:'test',
freeComment:'test',
};