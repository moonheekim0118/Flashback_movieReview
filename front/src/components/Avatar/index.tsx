import React from 'react';
import styled from 'styled-components';

interface Props {
    nickname?:string;
    imgSrc?:string;
    size?:number;
}

// 유저 아바타
const Avatar=({nickname, imgSrc="", size=40}:Props)=>{

    return(
        <Container size={size}>
            {imgSrc? <Image src={imgSrc}/> : <Nickname size={size}>{nickname[0]}</Nickname>}
        </Container>
    );
}

const Container = styled.div<{ size:number }>`
    width:${(props)=>`${props.size}px`};
    height:${(props)=>`${props.size}px`};
`;

const Image = styled.img`
    width:100%;
    height:100%;
    border-radius:100%;
    oveflow:hidden;
    object-fit:fill;
`;

const Nickname=styled.div<{ size:number }>`
    display:flex;
    justify-content:center;
    align-items:center;
    flex-direction:column;
    text-align:center;
    width:100%;
    height:100%;
    border-radius:100%;
    overflow:hidden;
    font-size:${(props)=>`${props.size-20}px`};
    color:#fff;
    background-color:#e0e0d1;
`;

export default Avatar;