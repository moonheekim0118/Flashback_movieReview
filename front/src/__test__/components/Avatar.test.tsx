import 'jsdom-global/register'; 
import React from "react";
import { MyInfoProp } from '../testUtils';
import { useSelector } from 'react-redux';
import { mount } from 'enzyme';
import Avatar from '../../components/Avatar';

jest.mock('react-redux', ()=>({
    useSelector : jest.fn(),
}));


describe('<Avatar/>',()=>{
    let container;
    beforeEach(()=>{
        useSelector.mockImplementation(()=>MyInfoProp);
        container = mount( <Avatar/> )
    })

    it('should match correctly', ()=>{
        expect(container.html()).toMatchSnapshot();
    })
});