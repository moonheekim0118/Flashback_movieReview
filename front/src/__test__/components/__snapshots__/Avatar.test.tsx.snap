// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> should match correctly 1`] = `"<div size=\\"50\\" class=\\"Avatar__Container-tewe7y-0 bkZJif\\"><div size=\\"50\\" class=\\"Avatar__Nickname-tewe7y-2 fFVjoY\\">t</div></div>"`;
